Bottom: c85cd283d15ca09eeb3035ff54cd6d5291a4e811
Top:    ad1f8d9e5e01a36e4eb34f7dc7d4de978cfb7f0b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 02:04:25 +0000

configure.ac: Use CPPFLAGS, not CFLAGS for our include directives.

We were previously using CFLAGS for adding pre-processor includes, where
these should really end up in the CPPFLAGS variable. Confusingly, their
constituents tend to come from variables set by autoconf / pkgconfig
expansions in the form of LIBNAME_CFLAGS.

Doing this allows the user to easily override CFLAGS at build time without
needing to replicate the enture include path list.


---

diff --git a/configure.ac b/configure.ac
index 5ea54c4..34eddb1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1106,9 +1106,9 @@ AC_DEFINE_UNQUOTED([COORD_TYPE],[$COORD_TYPE],
 AC_DEFINE_UNQUOTED([COORD_MAX],[$COORD_MAX],
   [Maximum value of coordinate type])
 
-# ------------- Complete set of CFLAGS and LIBS -------------------
+# ------------- Complete set of CPPFLAGS and LIBS -------------------
 
-CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GLIB_CFLAGS $GTK_CFLAGS $GD_CFLAGS $CAIRO_CFLAGS $GTKGLEXT_CFLAGS $GLU_CFLAGS $GL_CFLAGS"
+CPPFLAGS="$CPPFLAGS $X_CFLAGS $DBUS_CFLAGS $GLIB_CFLAGS $GTK_CFLAGS $GD_CFLAGS $CAIRO_CFLAGS $GTKGLEXT_CFLAGS $GLU_CFLAGS $GL_CFLAGS"
 LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GLIB_LIBS $GTK_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS $CAIRO_LIBS $GTKGLEXT_LIBS $GLU_LIBS $GL_LIBS"
